base/uk.ac.stfc.isis.ibex.ui.dae/src/uk/ac/stfc/isis/ibex/ui/dae/spectraplots/PersistedLocalChannel.java

```diff
@@ -80,4 +84,28 @@ public void onValue(T value) {
     public String getPVAddress() {
     	return address;
     }
+    
+    /**Attempt to write initial value to local PV. If fails (due to an existing race condition), try again until it passes.
+     * @param initialValue - value to be written to PV
+     */
+    private void write_to_pv(T initialValue) {
+    	Runnable task = new Runnable() {
+    		public void run() {
+    			int iterations = 1;
```

Comment:
`iterations` here is a local variable, meaning the check for `if iterations < 30` later on will always pass. I think we need to make this a field of this anonymous class instead. You can tell the current logic is not correct because if you make it look at `loc2://` rather than `loc://` the messages continue for more than 30s...

I would also declare it as `AtomicInteger` and use the `incrementAndGet`/`getAndIncrement` method as we are looking at this variable from a multithreaded context and `i++` is not an atomic operation. I don't think it would actually cause an issue in this specific scenario, but best to be safe in case we do refactoring later.